### PR TITLE
Add missing include and make libs shared

### PR DESCRIPTION
**Pull request by gerkey**
Redo of #203, where I did something wrong with git that made the ci job no longer work.

### CODE REVIEW

**Comment by gerkey** [comment]
First Windows CI attempt:

http://ci.ros2.org/job/ci_windows/2357/

**Comment by gerkey** [comment]
This time with the header actually committed:

http://ci.ros2.org/job/ci_windows/2362/

**Comment by Karsten1987** [comment]
I do believe you have to "activate" the symbol export within your `CMakeLists.txt`:

```
target_compile_definitions(tf2 PRIVATE "TF2_BUILDING_DLL")
```

**Comment by gerkey** [comment]
You mean that the preprocessor couldn't automatically figure out that that was my intent and just insert that definition for me? 😉

Here's the next attempt:

http://ci.ros2.org/job/ci_windows/2391/

**Comment by gerkey** [comment]
http://ci.ros2.org/job/ci_windows/2393/

File: tf2/include/tf2/LinearMath/MinMax.h [hunk 1]
```diff
@@ -18,25 +18,25 @@ subject to the following restrictions:
 #define GEN_MINMAX_H
 
 template <class T>
-TF2SIMD_FORCE_INLINE const T& tf2Min(const T& a, const T& b) 
+TF2_PUBLIC TF2SIMD_FORCE_INLINE const T& tf2Min(const T& a, const T& b)
```
Comment (Karsten1987) [inline tf2/include/tf2/LinearMath/MinMax.h]: You shouldn't need to export template symbols. Templates are getting evaluated on caller side, meaning they are always in the according compilation unit.

Comment (Karsten1987) [reply tf2/include/tf2/LinearMath/MinMax.h]: Also, Inline function are meant to be not exported.

Comment (gerkey) [reply tf2/include/tf2/LinearMath/MinMax.h]: Ah, that makes sense. Clearly I'm in unfamiliar territory here. So I should remove that declaration from all the template functions? Anywhere else?

Comment (gerkey) [reply tf2/include/tf2/LinearMath/MinMax.h]: I was reading about exporting and inline here:

https://msdn.microsoft.com/en-us/library/xa0d9ste.aspx

So they *can* be exported, but it's not clear to me whether they should or shouldn't be.

**Comment by gerkey** [comment]
http://ci.ros2.org/job/ci_windows/2394/

**Comment by gerkey** [comment]
http://ci.ros2.org/job/ci_windows/2395/

**Comment by gerkey** [comment]
~~~
19:12:49      9>LINK : fatal error LNK1181: cannot open input file 'Release\tf2_ros.lib' [C:\J\workspace\ci_windows\ws\build\tf2_ros\static_transform_publisher.vcxproj]
~~~

Looks like `tf2` is good and now `tf2_ros` needs the same visibility treatment. I'll get to that as soon as I can, maybe tomorrow. @Karsten1987, if you're blocked on this, feel free to jump in here and keep going without me.

**Comment by Karsten1987** [comment]
I thought I opened a PR with tf2_ros against your branch. Did you see it?

**Comment by gerkey** [comment]
Sorry, I hadn't noticed that. Thanks!

http://ci.ros2.org/job/ci_windows/2400/

**Comment by gerkey** [comment]
It builds!

Now we need to resolve compiler warnings:

http://ci.ros2.org/job/ci_windows/2400/warnings41Result/new/

**Comment by gerkey** [comment]
http://ci.ros2.org/job/ci_windows/2420/

**Comment by Karsten1987** [comment]
it looks to me as if the builds finally succeed. 
[Build Status](http://ci.ros2.org/job/ci_windows/2435/)

The build failure comes from two failing tests which are unrelated.

**Review by Karsten1987 — APPROVED** [review]

**Comment by gerkey** [comment]
Thanks for the all the help, @Karsten1987 !

**Pull request merged**